backend/sites/src/app/services/dropdown/dropdown.service.ts

```diff
@@ -60,80 +75,101 @@ export class DropdownService {
+   *
+   * @returns A promise that resolves to an array of objects with `metaData` and `dropdownDto` properties.
+   * @throws Error if there is an issue retrieving the data.
+   */
   async getNotationTypeCd() {
     try {
       const result = await this.eventTypeCdRepository.find();
       if (result) {
-        return result.reduce((acc, item) => {
-          const existingMetaData = acc.find(
-            (meta) => meta.metaData === item.eclsCode,
-          );
-          const dropdownItem = {
-            key: item.code,
-            value: item.description,
-          };
-          if (existingMetaData) {
-            existingMetaData.dropdownDto.push(dropdownItem);
-          } else {
-            acc.push({
-              metaData: item.eclsCode,
-              dropdownDto: [dropdownItem],
-            });
-          }
-
-          return acc;
-        }, []);
+        return result.reduce(
+          (acc, item: EventTypeCd) => {
+            const existingMetaData = acc.find(
+              (meta) => meta.metaData === item.eclsCode,
+            );
+            const dropdownItem = {
+              key: item.code,
+              value: item.description,
+            };
+            if (existingMetaData) {
+              existingMetaData.dropdownDto.push(dropdownItem);
+            } else {
+              acc.push({
+                metaData: item.eclsCode,
+                dropdownDto: [dropdownItem],
+              });
+            }
+            return acc;
+          },
+          [] as {
+            metaData: string;
+            dropdownDto: { key: string; value: string }[];
+          }[],
+        );
       }
+      return [];
     } catch (error) {
-      throw new Error('Failed to retrieve notation type code.');
+      throw new Error('Failed to retrieve notation type codes.');
     }
   }
 
+  /**
+   * Retrieves notation class codes from the repository.
+   *
+   * @returns A promise that resolves to an array of objects with `key` and `value` properties.
+   * @throws Error if there is an issue retrieving the data.
+   */
   async getNotationClassCd() {
     try {
       const result = await this.eventClassCdRepository.find();
       if (result) {
-        return result.map((obj: any) => ({
+        return result.map((obj: EventClassCd) => ({
           key: obj.code,
           value: obj.description,
         }));
       }
+      return [];
     } catch (error) {
-      throw new Error('Failed to retrieve notation class code.');
+      throw new Error('Failed to retrieve notation class codes.');
     }
   }
 
+  /**
+   * Retrieves notation participant role codes from the repository.
+   *
+   * @returns A promise that resolves to an array of objects with `key` and `value` properties.
+   * @throws Error if there is an issue retrieving the data.
+   */
   async getNotationParticipantRoleCd() {
     try {
       const result = await this.eventParticRoleCdRepository.find();
       if (result) {
```

Comment:
please have else block for this

Reply:
outside of block there is a return [] if there will be no result in any case it will return empty array, which is handled in resolver